src/plugins/recursion/README.md

```diff
@@ -81,10 +81,186 @@ kdb set /recursive/define/#2/ref/#1 /recursive/define/#5
+kdb set /editor/menu/#8 "Organize Macros"
+kdb set /editor/menu/#9 "Macro Basics"
+```
+
+Now that we have defined we can link together all menues. Lets start with the `File` menu:
+
+```
+kdb set /editor/menu/#0/menuref/#0 /editor/menu/#2
+kdb set /editor/menu/#0/menuref/#1 /editor/menu/#3
+```
+
+Next we link the `Settings` menu:
+
+```
+kdb set /editor/menu/#3/menuref/#0 /editor/menu/#5
+kdb set /editor/menu/#3/menuref/#1 /editor/menu/#6
+```
+
+Now we link the `Tools` section:
+```
+kdb set /editor/menu/#1/menuref/#0 /editor/menu/#5
+kdb set /editor/menu/#1/menuref/#1 /editor/menu/#3
+kdb set /editor/menu/#1/menuref/#2 /editor/menu/#7
+```
+
+Note that `Settings` is already linked. We do not need to do it for the `Tools` menu again.
+
+Finally we link all submenus of `Macros`:
+
+```
+kdb set /editor/menu/#7/menuref/#0 /editor/menu/#8
+kdb set /editor/menu/#8/menuref/#0 /editor/menu/#9
+```
+
+### Example of wrong settings
+
+In the above example the following settings will be rejected:
+
+```
+#Link "Macro Basics" to the parent menu "Macros" and create an endless loop
+kdb set /editor/menu/#9/menuref/#0 /editor/menu/#7
+ERR: 198
+Reason: Cyclical reference detected: "Macro Basics" -> "Macros" -> Organize Macros" -> "Macro Basics"
+
+#Link to something which does not exist
+kdb set /editor/menu/#9/menuref/#0 /editor/menu/#10
+ERR: 199
+Reason: Could not find vertex with #10. Maybe it does not exist yet?
+```
+
+Sitenote: Maybe the last example here will just emit a warning so users are free in the order of declaration.
+
+### Problems
+
+#### Klemens architecture
+Klemens exported the [ini file](https://github.com/Piankero/lcdproc-specification/blob/master/src/lcdexec/misc/lcdexec_elektra.ini) 
+and showed me his approach.
+
+First of all in his approach, he differs the `main` menu and the `menu` which are below the main menu differently. Also they have an additional attribute
+for the displayname.
+
+In the above example, the LCDproc variant would look like this in the current version:
+```
+kdb set /editor/menu/main/#0/displayname File
+kdb set /editor/menu/main/#1/displayname Tools
+```
+
+Additionally they refer to submenu's implicitly. Eg the `Print` and `Settings` menu would be set like this:
+
+```
+kdb set /editor/menu/main/#0/menu/#0/displayname Print
+kdb set /editor/menu/main/#0/menu/#1/displayname Settings
+```
+
+Note the difference between `main` and `menu` in the key. Additionally it would prohibit the reuse of the
+`Settings` substructure for the `Tools` menu. The user would need to define everything again instead of just
+referencing to it.
+
+#### LCDproc's current format
+
+[This](https://github.com/lcdproc/lcdproc/blob/master/clients/lcdexec/lcdexec.conf) is the current format in
+lcdexec.
+
+Their approach is not array based but name based. 
+Their reference keyword (`check/recursion/ref`) to other settings is `Entry`.
+
+They do not use an array based approach though but give names instead.
```

Comment:
Can we add a name-based approach or does it confict with the array-based approach?

Reply:
It doesn't conflicht imo, as discussed in your comment below it needs to know if it is a vertex or some arbitrary setting unrelated to the recursive structure. With the array based approach this is clear.

Reply:
Imho, the array does not make a difference. Without some external knowledge you never know if a string is a reference or a value interpreted otherwise.